tools/c7n_azure/c7n_azure/resources/sqldatabase.py

```diff
@@ -238,3 +271,174 @@ def get_retention_from_backup_policy(self, retention_policy):
         if actual_duration_units.iso8601_symbol != self.retention_period_units.iso8601_symbol:
             return None
         return actual_duration
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BackupRetentionPolicyAction(AzureBaseAction):
+
+    def __init__(self, operations_property, *args, **kwargs):
+        super(BackupRetentionPolicyAction, self).__init__(*args, **kwargs)
+        self.operations_property = operations_property
+
+    def _process_resource(self, database):
+        client = self.manager.get_client()
+        update_operation = getattr(client, self.operations_property).create_or_update
+
+        resource_group_name, server_name, database_name = \
+            BackupRetentionPolicyHelper.get_backup_retention_policy_context(database)
+        parameters = self.get_parameters_for_new_retention_policy(database)
+
+        update_operation(resource_group_name, server_name, database_name, parameters).result()
+
+    @abc.abstractmethod
+    def get_parameters_for_new_retention_policy(self, database):
```

Comment:
`_get_parameters_for_new_retention_policy` (`_` to mark internal methods)

Reply:
should this be done even for methods that are intended to be overridden by an inheriting class?

Reply:
I think so.. This is not a public interface, and the fact that is needs to be implemented via inheriting is an implementation detail.